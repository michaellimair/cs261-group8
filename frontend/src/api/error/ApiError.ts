class ApiError<T> extends Error {
  data?: T;

  constructor(message: string, data?: T) {
    super(message ?? 'An error has occurred in the API.');
    this.data = data;
  }

  static isApiError(obj: any): obj is ApiError<any> {
    return obj instanceof ApiError;
  }
}

export default ApiError;
