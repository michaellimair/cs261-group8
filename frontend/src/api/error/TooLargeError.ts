import ApiError from './ApiError';

class TooLargeError<T extends any> extends ApiError<T> {
  constructor(message?: string, data?: T) {
    super(message ?? 'File too large', data);
  }

  static isTooLargeError<T>(obj: any): obj is TooLargeError<T> {
    return obj instanceof TooLargeError;
  }
}

export default TooLargeError;
