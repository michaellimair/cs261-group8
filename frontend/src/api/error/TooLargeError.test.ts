import ApiError from './ApiError';
import TooLargeError from './TooLargeError';

describe('TooLargeError', () => {
  it('is instantiated successfully', () => {
    interface ISampleError {
      status_code: number;
      message: string;
    }

    const errorData = {
      status_code: 400,
      message: 'An error has occurred',
      errors: [],
    };

    const err = new TooLargeError<ISampleError>(errorData.message, errorData);

    expect(err.message).toBe(errorData.message);
    expect(err.data).toMatchObject(errorData);
    expect(err).toMatchSnapshot();
  });

  it('is a subclass of ApiError', () => {
    const err = new TooLargeError<unknown>('err', {});

    expect(ApiError.isApiError(err)).toBe(true);
  });

  describe('isTooLargeError', () => {
    it('identifies TooLargeError classes correctly', () => {
      const err = new TooLargeError<unknown>('err', {});

      expect(TooLargeError.isTooLargeError(err)).toBe(true);
    });
  });
});
