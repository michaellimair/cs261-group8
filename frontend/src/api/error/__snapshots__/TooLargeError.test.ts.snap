// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooLargeError is instantiated successfully 1`] = `[Error: An error has occurred]`;
